src/utils/src/arg_parser.rs

```diff
@@ -48,59 +48,36 @@ impl fmt::Display for Error {
     }
 }
 
-/// Keep information about the application.
+/// Keep information about the argument parser.
 #[derive(Clone, Default)]
-pub struct App<'a> {
+pub struct ArgParser<'a> {
     arguments: Arguments<'a>,
-    name: &'a str,
-    version: &'a str,
-    header: &'a str,
 }
 
-impl<'a> App<'a> {
-    /// Add an argument with its associated `ArgInfo` in `arguments`.
-    pub fn arg(mut self, arg_info: ArgInfo<'a>) -> Self {
-        self.arguments.insert_arg(arg_info);
-        self
-    }
-
-    /// Create a new App instance.
+impl<'a> ArgParser<'a> {
+    /// Create a new ArgParser instance.
     pub fn new() -> Self {
-        App::default()
+        ArgParser::default()
     }
 
-    /// Set the name of the running application.
-    pub fn name(mut self, name: &'a str) -> Self {
-        self.name = name;
-        self
-    }
-
-    /// Set the version of the running application.
-    pub fn version(mut self, version: &'a str) -> Self {
-        self.version = version;
-        self
-    }
-
-    /// Set general information about the running application.
-    pub fn header(mut self, header: &'a str) -> Self {
-        self.header = header;
+    /// Add an argument with its associated `Argument` in `arguments`.
+    pub fn arg(mut self, argument: Argument<'a>) -> Self {
+        self.arguments.insert_arg(argument);
         self
     }
 
     /// Parse the command line arguments.
-    pub fn parse_cmdline_args(&mut self) -> Result<()> {
+    pub fn parse_from_cmdline(&mut self) -> Result<()> {
         self.arguments.parse_from_cmdline()
     }
 
     /// Concatenate the `help` information of every possible argument
     /// in a message that represents the correct command line usage
     /// for the application.
```

Comment:
I would've expected the banner to be optional and not related to the ArgumentParser at all. For example, users could print the `header` before calling `formatted_help` if they want to.

In any case, if you want to keep it as is, please update the documentation.

Reply:
You're right, it makes more sense to be separated.